File: iosApp/src/Preview/Factories/InboxItemFactory.swift
import Foundation

#if DEBUG

enum InboxItemFactory {

    // MARK: - Named Variants

    /// Unread invitation — action required, not yet read.
    static var unreadInvitation: InboxItemModel {
        InboxItemModel(
            id: "inbox-unread-inv-001",
            title: "You're invited to Week-end ski 2024",
            message: "Marie Dupont invited you to join the trip",
            timeAgo: "17m",
            type: .invitation,
            isRead: false,
            commentCount: 0,
            isFocused: true,
            eventName: "Week-end ski 2024",
            eventId: "evt-inv-001"
        )
    }

    /// Read update — poll update type, informational, already read.
    static var readUpdate: InboxItemModel {
        InboxItemModel(
            id: "inbox-read-upd-002",
            title: "Poll results updated",
            message: "3 new votes on Birthday Dinner",
            timeAgo: "3h",
            type: .pollUpdate,
            isRead: true,
            commentCount: 5,
            isFocused: false,
            eventName: "Birthday Dinner",
            eventId: "evt-upd-002"
        )
    }

    /// Action required — vote reminder, unread.
    static var actionRequired: InboxItemModel {
        InboxItemModel(
            id: "inbox-action-003",
            title: "Vote reminder: Team Offsite",
            message: "Deadline is tomorrow — cast your vote now",
            timeAgo: "1h",
            type: .pollUpdate,
            isRead: false,
            commentCount: 2,
            isFocused: true,
            eventName: "Team Offsite",
            eventId: "evt-action-003"
        )
    }

    /// Completed — event confirmed, success status, already read.
    static var completed: InboxItemModel {
        InboxItemModel(
            id: "inbox-done-004",
            title: "Event confirmed: Summer BBQ",
            message: "Saturday 14 June at 16:00 — see you there!",
            timeAgo: "1d",
            type: .eventUpdate,
            isRead: true,
            commentCount: 12,
            isFocused: false,
            eventName: "Summer BBQ",
            eventId: "evt-done-004"
        )
    }

    // MARK: - Builder

    /// Create an inbox item with sensible defaults. Override only the parameters you need.
    static func make(
        id: String = "inbox-\(UUID().uuidString.prefix(8))",
        title: String = "Test notification",
        message: String = "Something happened in your event",
        timeAgo: String = "5m",
        type: InboxItemType = .invitation,
        isRead: Bool = false,
        commentCount: Int = 0,
        isFocused: Bool = false,
        eventName: String? = nil,
        eventId: String? = nil
    ) -> InboxItemModel {
        InboxItemModel(
            id: id,
            title: title,
            message: message,
            timeAgo: timeAgo,
            type: type,
            isRead: isRead,
            commentCount: commentCount,
            isFocused: isFocused,
            eventName: eventName,
            eventId: eventId
        )
    }

    // MARK: - Collections

    /// Generates a mixed list of inbox items cycling through the named variants.
    static func mixedList(count: Int = 8) -> [InboxItemModel] {
        let templates: [() -> InboxItemModel] = [
            { unreadInvitation },
            { readUpdate },
            { actionRequired },
            { completed }
        ]

        return (0..<count).map { index in
            var item = templates[index % templates.count]()
            item.title = "\(item.title) (\(index + 1))"
            return InboxItemModel(
                id: "inbox-mixed-\(index)",
                title: item.title,
                message: item.message,
                timeAgo: item.timeAgo,
                type: item.type,
                isRead: item.isRead,
                commentCount: item.commentCount,
                isFocused: item.isFocused,
                eventName: item.eventName,
                eventId: item.eventId
            )
        }
    }
}

#endif

File: composeApp/src/androidMain/kotlin/com/guyghost/wakeve/preview/factories/InboxItemFactory.kt
package com.guyghost.wakeve.preview.factories

import com.guyghost.wakeve.models.InboxItem
import com.guyghost.wakeve.models.InboxItemStatus
import com.guyghost.wakeve.models.InboxItemType

/**
 * Factory for creating preview/test InboxItem instances.
 *
 * All properties use `get()` to return fresh instances on each access.
 */
object InboxItemFactory {

    /** An unread event invitation requiring RSVP. */
    val unreadInvitation
        get() = InboxItem(
            id = "inbox-001",
            type = InboxItemType.EVENT_INVITATION,
            status = InboxItemStatus.ACTION_REQUIRED,
            title = "You're invited to Weekend Hiking Trip",
            subtitle = "Marie Dupont invited you",
            description = "A two-day hike in the Vosges mountains. Please vote on your preferred dates.",
            eventId = "event-confirmed-002",
            eventTitle = "Weekend Hiking Trip",
            timestamp = "2026-03-03T09:00:00Z",
            isRead = false,
            commentCount = 0
        )

    /** A read informational update about poll results. */
    val readUpdate
        get() = InboxItem(
            id = "inbox-002",
            type = InboxItemType.POLL_UPDATE,
            status = InboxItemStatus.INFO,
            title = "Poll results updated",
            subtitle = "Team Offsite Q2",
            description = "3 new votes have been submitted. April 12 is currently leading.",
            eventId = "event-polling-003",
            eventTitle = "Team Offsite Q2",
            timestamp = "2026-03-02T15:30:00Z",
            isRead = true,
            commentCount = 2
        )

    /** An unread item that requires the user to take action (vote reminder). */
    val actionRequired
        get() = InboxItem(
            id = "inbox-003",
            type = InboxItemType.VOTE_REMINDER,
            status = InboxItemStatus.ACTION_REQUIRED,
            title = "Reminder: vote before April 8",
            subtitle = "Team Offsite Q2",
            description = "The voting deadline is approaching. Please submit your availability.",
            eventId = "event-polling-003",
            eventTitle = "Team Offsite Q2",
            timestamp = "2026-03-03T08:00:00Z",
            isRead = false,
            commentCount = 0
        )

    /** A completed / closed notification about a confirmed event. */
    val completed
        get() = InboxItem(
            id = "inbox-004",
            type = InboxItemType.EVENT_CONFIRMED,
            status = InboxItemStatus.COMPLETED,
            title = "Date confirmed: December 20",
            subtitle = "Holiday Dinner 2025",
            description = "The date has been finalized. Check the event for details.",
            eventId = "event-past-004",
            eventTitle = "Holiday Dinner 2025",
            timestamp = "2025-12-16T10:00:00Z",
            isRead = true,
            commentCount = 5
        )

    /** A success notification when a new participant joins. */
    val participantJoined
        get() = InboxItem(
            id = "inbox-005",
            type = InboxItemType.PARTICIPANT_JOINED,
            status = InboxItemStatus.SUCCESS,
            title = "Claire Bernard joined",
            subtitle = "Weekend Hiking Trip",
            description = "A new participant has joined your event.",
            eventId = "event-confirmed-002",
            eventTitle = "Weekend Hiking Trip",
            timestamp = "2026-03-02T12:00:00Z",
            isRead = true,
            commentCount = 0
        )

    /** A warning notification about a budget update. */
    val budgetWarning
        get() = InboxItem(
            id = "inbox-006",
            type = InboxItemType.BUDGET_UPDATE,
            status = InboxItemStatus.WARNING,
            title = "Budget exceeded by 15%",
            subtitle = "Team Offsite Q2",
            description = "The estimated cost now exceeds the planned budget.",
            eventId = "event-polling-003",
            eventTitle = "Team Offsite Q2",
            timestamp = "2026-03-01T16:45:00Z",
            isRead = false,
            commentCount = 1
        )

    /**
     * Returns a mixed list of inbox items covering various types and statuses.
     *
     * @param count Number of items to return (default 6, max cycles through all templates).
     */
    fun mixedList(count: Int = 6): List<InboxItem> {
        val all = listOf(
            unreadInvitation,
            readUpdate,
            actionRequired,
            completed,
            participantJoined,
            budgetWarning
        )
        if (count <= all.size) return all.take(count)

        // Cycle through templates for larger lists, generating unique IDs
        return (0 until count).map { index ->
            val template = all[index % all.size]
            template.copy(id = "inbox-mixed-${index.toString().padStart(3, '0')}")
        }
    }
}
